packages/joint-core/test/utils.js

```diff
@@ -88,14 +88,135 @@
 
     function normalizeCssAttr(name, value) {
 
-        var $tmpEl = $('<div/>').appendTo($('body'));
-        var normalizedValue = $tmpEl.css(name, value).css(name);
-        $tmpEl.remove();
+        var tmpEl = document.createElement('div');
+        document.body.appendChild(tmpEl);
+        tmpEl.style[name] = value;
+        var normalizedValue = tmpEl.style[name];
+        document.body.removeChild(tmpEl);
         return normalizedValue;
     }
 
 })(QUnit.assert);
 
+// Dom manipulation helpers.
+// -------------------------
+
+const $ = window.$ = joint.mvc.$;
+
+$.contains = function(parent, node) {
+    return parent !== node && parent.contains(node);
+};
+
+function matchFilter(elem, filter) {
+    if (typeof filter === 'string') {
+        if (!elem.matches(filter)) return false;
+    } else if (typeof filter === 'function') {
+        if (!filter(elem)) return false;
+    } else if (filter === 'object') {
+        if (elem !== filter) return false;
+    }
+    return true;
+}
+
+function dir(elem, dir, filter) {
+    var sibs = [];
+    while ((elem = elem[dir])) {
+        if (!matchFilter(elem, filter)) continue;
+        sibs.push(elem);
+    }
+    return sibs;
+}
+
+function sibling(elem, dir, filter) {
+    const prevEl = elem[dir];
+    if (!matchFilter(elem, filter)) return $();
+    return $(prevEl);
+}
+
+$.fn.prevAll = function(filter) {
+    const [el] = this;
+    return $(dir(el, 'previousElementSibling', filter));
+};
+
+$.fn.nextAll = function(filter) {
+    const [el] = this;
+    return $(dir(el, 'nextElementSibling', filter));
+};
+
+$.fn.prev = function(filter) {
+    const [el] = this;
+    return $(sibling(el, 'previousElementSibling', filter));
+};
+
+$.fn.index = function() {
+    const [el] = this;
+    return Array.prototype.indexOf.call(el.parentNode.children, el);
+};
+
+$.fn.offset = function() {
+    const [el] = this;
+    const box = el.getBoundingClientRect();
+    return {
+        top: box.top + window.pageYOffset - document.documentElement.clientTop,
```

Comment:
`pageYOffset` and `pageXOffset` are deprecated according to VS Code, suggests changing to `scrollY` and `scrollX`